test/Confluent.Kafka.IntegrationTests/Tests/Producer_CustomPartitioner.cs

```diff
@@ -117,6 +117,11 @@ public void Producer_CustomPartitioner(string bootstrapServers)
 
 
             // Null key
+            producerConfig = new ProducerConfig
+            {
+                BootstrapServers = bootstrapServers,
+                StickyPartitioningLingerMs = 0 // disable sticky partitioning and run the default partitioner for null keys.
+            };
```

Comment:
The original test should've failed by also asserting that the partitioner was called, so I recommend adding that too, i.e., a counter in the partitioner that is checked after Flush to be equal to the message count (1).

Reply:
Thanks, updated to include this.